lib/sparse/src/common/sparse_vector.rs

```diff
@@ -97,14 +102,41 @@ impl RemappedSparseVector {
         self.indices.windows(2).all(|w| w[0] < w[1])
     }
 
+    /// Score this vector against another vector using dot product.
+    /// Warning: Expects both vectors to be sorted by indices.
+    ///
+    /// Return None if the vectors do not overlap.
+    pub fn score_measured(
+        &self,
+        other: &RemappedSparseVector,
+        hardware_counter: &HardwareCounterCell,
+    ) -> Option<ScoreType> {
+        debug_assert!(self.is_sorted());
+        debug_assert!(other.is_sorted());
+        score_vectors(
+            &self.indices,
+            &self.values,
+            &other.indices,
+            &other.values,
+            hardware_counter,
+        )
+    }
+
     /// Score this vector against another vector using dot product.
     /// Warning: Expects both vectors to be sorted by indices.
     ///
     /// Return None if the vectors do not overlap.
     pub fn score(&self, other: &RemappedSparseVector) -> Option<ScoreType> {
         debug_assert!(self.is_sorted());
         debug_assert!(other.is_sorted());
-        score_vectors(&self.indices, &self.values, &other.indices, &other.values)
+        let hardware_counter = HardwareCounterCell::new();
```

Comment:
I'm not sure which is the best practical way of handling this properly.
Having the score function twice, one metered and one not is unpractical in my opinion. However measuring and throwing the measurements away like here isn't clean either.
Feel free to suggest which approach is preferred

Reply:
Why exactly do we need unmeasured function?

Reply:
Because [here](https://github.com/qdrant/qdrant/blob/dev/lib/sparse/src/index/search_context.rs#L131) we already do measurement manually and don't need the `score` function to measure anything.
[Here](https://github.com/qdrant/qdrant/blob/more_scorer_measurements/lib/segment/src/vector_storage/query_scorer/sparse_custom_query_scorer.rs#L51) we do need the score function to measure though.